plugin/storage/grpc/shared/grpc_client.go

```diff
@@ -52,11 +54,45 @@ type grpcClient struct {
 func upgradeContextWithBearerToken(ctx context.Context) context.Context {
 	bearerToken, hasToken := spanstore.GetBearerToken(ctx)
 	if hasToken {
-		requestMetadata := metadata.New(map[string]string{
-			spanstore.BearerTokenKey: bearerToken,
-		})
-		return metadata.NewOutgoingContext(ctx, requestMetadata)
+		md, ok := metadata.FromOutgoingContext(ctx)
+		if !ok {
+			md = metadata.New(nil)
+		} else {
+			md = md.Copy()
```

Comment:
Why do you need a copy?

Reply:
Removed. I've checked it is unnecessary.